site/blog/2016/jul/1/sphinx-and-rtd-for-writers.rst

```diff
@@ -0,0 +1,548 @@
+
+       import antigravity
+
+       def main():
+           antigravity.fly()
+
+This would display the Python snippet,
+syntax highlighted,
+with line numbers.
+The ``python`` in the above example is an argument to the ``code-block`` *directive*.
+The ``:linesnos:`` is an *option* that says to display line numbers for the code block.
+In the HTML output,
+it would look like this:
+
+.. image:: code.png
+
+Sphinx doesn't stop there though.
+It allows you to store your code examples in external files,
+and be included in your documentation for easier maintenance.
+This uses the ``literalinclude`` directive::
+
+    .. literalinclude:: example.rb
+       :language: ruby
+       :emphasize-lines: 12,15-18
+
+The neat thing here is the ``:emphasize-lines:``.
+This shows the lines highlighted in your output.
+This is quite useful for showing sets of code examples where a subset of the code changes.
+You can also specify just a subset of lines to show with the ``:lines:`` option,
+so you don't have to manage multiple snippets.
+
+These directives have a lot more powerful options than I can show in this article.
+You can see the full documentation for them on the Sphinx website:
+
+* http://www.sphinx-doc.org/en/stable/markup/code.html
+
+Working with References
+```````````````````````
+
+A powerful reference system is a large part of the power of Sphinx.
+You are able to reference arbitrary headings and paragraphs in your content,
+but also semantically reference a large number of things as well.
+On top of that,
+Sphinx includes ``intersphinx`` which allows referencing across Sphinx projects.
+This means that if you have multiple projects inside your company,
+you can still use semantic referencing across them!
+
+A simple reference is defined as such::
+
+    .. _reference-target-name::
+
+    This is a bit of content.
+
+    This is how you point to the above reference, :ref:`reference-target-name`
+
+Sphinx also includes a number of other semantic reference types.
+Here are a few examples:
+
+* ``:doc:`` for referencing documents
+* ``:cls:`` for referencing programming classes
+* ``:term:`` for referencing glosary terms
+
+All references support ``intersphinx``,
+which allows you to prefix your references with a specific project name.
+So if your user guide needs to reference your API documentation,
+you could do::
+
+    Check out the :cls:`api-ref:api.request`
+
+Which would know to look in your ``api-ref`` project for the class.
+The name ``api-ref`` is arbitrary,
+and is defined in your project configuration's ``intersphinx_mapping``.
+
+You see the complete set of references in the Sphinx documentation:
+
+* Cross Referencing Syntax - http://www.sphinx-doc.org/en/stable/markup/inline.html#cross-referencing-syntax
+* Intersphinx - http://www.sphinx-doc.org/en/stable/ext/intersphinx.html
+
+Including comments form source code
+```````````````````````````````````
+Integration with software is one of the largest benefits of Sphinx.
```

Comment:
I think I'd prefer "code" to "software" here (clearer and perhaps more comprehensible to writer audience). Also, I don't think it matters as much as it used to, but the MS critter is .NET, not .Net.